nixos/modules/services/hardware/display.nix

```diff
@@ -0,0 +1,165 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.hardware.display;
+in
+{
+  options = {
+    hardware.display.edid.packages = lib.mkOption {
+      type = with lib.types; listOf package;
+      default = [ ];
+      description = lib.mdDoc ''
+        List of packages containing EDID binary files at `$out/lib/firmware/edid`.
+        Such files will be available for use in `drm.edid_firmware` kernel
+        parameter as `edid/<filename>`.
+
+        You can craft one directly here or use sibling options `linuxhw` and `modelines`.
+      '';
+      example = lib.literalExpression ''
+        [
+          (pkgs.runCommand "edid-custom" {} '''
+            mkdir -p "$out/lib/firmware/edid"
+            base64 -d > "$out/lib/firmware/edid/custom1.bin" <<'EOF'
+            <insert your base64 encoded EDID file here `base64 < /sys/class/drm/card0-.../edid`>
+            EOF
+          ''')
+        ]
+      '';
+      apply = list:
+        if list == [ ] then null else
+        (pkgs.buildEnv {
+          name = "firmware-edid";
+          paths = list;
+          pathsToLink = [ "/lib/firmware/edid" ];
+          ignoreCollisions = true;
+        }) // {
+          compressFirmware = false;
+        };
+    };
+
+    hardware.display.edid.linuxhw = lib.mkOption {
+      type = with lib.types; attrsOf (listOf str);
+      default = { };
+      description = lib.mdDoc ''
+        Exposes EDID files from https://github.com/linuxhw/EDID
```

Comment:
As an end user, my first question when looking at this would be when to use it. Basically, how can I tell whether my system already has the relevant EDID file *somewhere*? I'm not certain whether this is the right place to put this documentation verbatim, but a link to a good reference would also work.

Reply:
I'm pretty sure the link contains enough background information, but I can add some practical information to the docs.